/*
 * Copyright (C) 2017 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

package profiler.proto;
option java_package = "com.android.tools.profiler.proto";
option java_outer_classname = "EnergyProfiler";

import "common.proto";

service EnergyService {

  // Requests energy data from app with given |process_id|, in the time range
  // from |start_timestamp| (exclusive) to |end_timestamp| (inclusive), or
  // mathematically written as in interval (start_timestamp, end_timestamp].
  rpc GetData(EnergyDataRequest) returns (EnergyDataResponse) {
  }
}

message EnergyDataRequest {
  Session session = 1;
  int64 start_timestamp = 2;
  int64 end_timestamp = 3;
}

message EnergyDataResponse {
  message EnergySample {
    // Time when the energy sample usages occured.
    int64 timestamp = 1;
    // Sample usage data by cpu in mAh.
    int32 cpu_usage = 2;
    // Sample usage data by network in mAh.
    int32 network_usage = 3;
  }

  // List of Energy usage sample data.
  repeated EnergySample sample_data = 1;
}
